torch/distributed/_tensor/random.py

```diff
@@ -142,3 +143,73 @@ def _set_offset(new_offset: int, device_mesh: DeviceMesh) -> None:
                 f"DTensor randomness only supports cuda device type, "
                 f"but got {device_mesh.device_type}"
             )
+
+
+def _calc_start_offset(spec: DTensorSpec) -> int:
```

Comment:
see my above comments, we don't really need to return anything, in this function itself, we should get the current rng state, calculate offset then set the rng state.
Similarly it applies to the "post_process_rng"